src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,175 @@
+        /// </summary>
+        /// <paramref name="i2cDevice">Reference to the initialized I2C interface device</paramref>
+        public AhtBase(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+
+            // even if not clearly stated in datasheet, start with a software reset to assure clear start conditions
+            SoftReset();
+
+            // check whether the device indicates the need for a calibration cycle
+            // and perform calibration if indicated ==> c.f. datasheet, version 1.1, ch. 5.4
+            if (!IsCalibrated())
+            {
+                Calibrate();
+            }
+        }
+
+        /// <summary>
+        /// Gets the current temperature reading from the sensor.
+        /// Reading the temperature takes between 10 ms and 80 ms.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetTemperature()
+        {
+            Measure();
+            return Temperature.FromDegreesCelsius(_temperature);
+        }
+
+        /// <summary>
+        /// Gets the current humidity reading from the sensor.
+        /// Reading the humidity takes between 10 ms and 80 ms.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Ratio GetHumidity()
+        {
+            Measure();
+            return Ratio.FromPercent(_humidity);
+        }
+
+        /// <summary>
+        /// Perform sequence to retrieve current readings from device
+        /// </summary>
+        private void Measure()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                (byte)Command.Measure,
+                0x33,
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+
+            // According to the datasheet the measurement takes 80 ms and completion is indicated by the status bit.
+            // However, it seems to be faster at around 10 ms and sometimes up to 50 ms.
+            while (IsBusy())
+            {
+                Thread.Sleep(10);
+            }
+
+            buffer = stackalloc byte[6];
+            _i2cDevice.Read(buffer);
+
+            // data format: 20 bit humidity, 20 bit temperature
+            // 7               0 7              0 7             4           0 7          0 7         0
+            // [humidity 19..12] [humidity 11..4] [humidity 3..0|temp 19..16] [temp 15..8] [temp 7..0]
+            // c.f. datasheet ch. 5.4.5
+            Int32 rawHumidity = (buffer[1] << 12) | (buffer[2] << 4) | (buffer[3] >> 4);
+            Int32 rawTemperature = ((buffer[3] & 0xF) << 16) | (buffer[4] << 8) | buffer[5];
+            // RH[%] = Hraw / 2^20 * 100%, c.f. datasheet ch. 6.1
+            _humidity = (rawHumidity * 100.0) / 0x100000;
```

Comment:
can you create Ratio directly here? Is it possible to avoid multiplying by floating point number?

Reply:
Maybe...but for the sake of clarity and relation to the datasheet I would like to stick to this. 
Is there anything wrong with using the floating point here? Precision shouldn't be an issue.